### PR TITLE
Add EVM documentation

### PR DESCRIPTION
**Pull request by mikedotexe**
When this PR is submitted an "onrender" site will spin up so the documentation can be easily navigated with the code of this branch.
When the URL is available, please click it. Then navigate by:
1. at the top "develop"
2. on the left sidebar, scroll down to Ethereum Compatibility

Then you may review the three items added there; they're all new.

### CODE REVIEW

File: docs/evm/near-web3-provider.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  keyStore: new nearAPI.keyStores.BrowserLocalStorageKeyStore(),
+  isReadOnly: true
+});
+```
+
+###  Instantiating signed
+
+For the signed instance, this assumes that the browser has local storage containing a key to the account specified.
+
+For folks unfamiliar with NEAR Wallet, a brief recap as the workflow in NEAR is different than that of a browser extension or by scanning QR codes. Let's imagine a user wants to use a NEAR betanet account to interact with the EVM. The first step is to create an account with Wallet. (Non-mainnet networks allow for free account creation with an initial balancer of Ⓝ.)
```
Comment (artob) [inline docs/evm/near-web3-provider.md:45]: "initial balancer of Ⓝ"

File: docs/evm/near-web3-provider.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+NEAR Betanet Wallet:
+
+https://wallet.betanet.near.org
+
+At this point, the NEAR Wallet website's local storage contains a key for this account. When you want to interact with a frontend using your account, you'll typically have the frontend redirect to Wallet, requesting the creation a special function-call access key. After allowing the creation, the browser redirects back to the dApp frontend where local storage contains this new key.
```
Comment (artob) [inline docs/evm/near-web3-provider.md:51]: "where local storage contains this new key" => "with local storage containing this new key"

File: docs/evm/near-web3-provider.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+This covers the basic usage of `near-web3-provider` on the frontend, allowing your Ethereum smart contracts to interact with a website.
+
+## Backend
+
+The [NEAR Pet Shop example](https://github.com/near-examples/near-pet-shop) demonstrates both of these. Please see the [Truffle page](/evm/truffle) for details on the backend approach.
+
+### NEAR CLI
+
+You can interact with an EVM smart contract with NEAR CLI, the command-line interface tool. With NodeJS version 12+, you may install this globally with:
```
Comment (artob) [inline docs/evm/near-web3-provider.md:104]: "with NEAR CLI" => "with the NEAR CLI"

File: docs/evm/near-web3-provider.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    
+For more information see the [NEAR CLI documentation](/docs/development/near-cli).
+
+Be sure to have followed the instructions earlier on creating a betanet NEAR account with Wallet. Issue this command to create a full-access key file locally: 
+
+    env NEAR_ENV=betanet near login
+    
+The key file will be located at `~/.near-credentials/betanet/yourname.betanet.json`.
+
+Now we're ready to use NEAR CLI to interact with a smart contract on the NEAR EVM. Let's say we've used Truffle to build and deploy a smart contract as [detailed in this guide](/evm/truffle#build-and-deploy), and the **contract address** is output as `0xAdf11a39283CEB00DEB90a5cE9220F89c6C27E67`. In this case the contract is NEAR Pet Shop example mentioned earlier. There is a read-only method `getAdopters` and a method that changes state called `adopt`.
```
Comment (artob) [inline docs/evm/near-web3-provider.md:116]: "NEAR CLI" => "the NEAR CLI"

File: docs/evm/truffle.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
+---
+id: truffle
+title: Using Truffle with the EVM
+sidebar_label: Using Truffle
+---
+
+[Truffle](https://trufflesuite.com) is a popular tool for EVM-based smart contract development and testing. A typical dApp using Truffle will have a file called `truffle-config.js` in the project root specifying details on connection, compilation, networks, and so on.
```
Comment (artob) [inline docs/evm/truffle.md:7]: The actual tool is at https://www.trufflesuite.com/truffle, the whole tool suite at https://www.trufflesuite.com.

File: docs/evm/truffle.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+NEAR has a custom provider called [`near-web3-provider`](https://github.com/near/near-web3-provider) that can be specified in truffle configuration. It's published as an [NPM package](https://npmjs.com/package/near-web3-provider) and can be added with:
```
Comment (artob) [inline docs/evm/truffle.md:9]: "truffle" => "Truffle"

File: docs/evm/truffle.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+To see an example of this in a project, see the [NEAR Pet Shop example](https://github.com/near-examples/near-pet-shop/blob/master/truffle-config.js).
+
+## Testing
+
+At this time, tests run using an existing account on the network. For example, if a user is going to test on betanet, they would first create a betanet account at:
+
+https://wallet.betanet.near.org
+
+Then use [NEAR CLI](/docs/development/near-cli) to login to that account, which creates a full-access key file in their home directory. (Specifically `~/.near-credentials`) Now tools and libraries like NEAR CLI and [`near-api-js`](https://github.com/near/near-api-js) can utilize these keys for actions including deploying contracts, transferring Ⓝ, creating subaccounts, etc. (See all available [Actions here](https://nomicon.io/RuntimeSpec/Actions.html).) The accounts created for testing are subaccounts (ex. `subaccount.yourname.betanet` created from `yourname.betanet`)
```
Comment (artob) [inline docs/evm/truffle.md:71]: "NEAR CLI" => "the NEAR CLI". (I get the feeling you're using this is as a proper name? That'd be clearer if more clearly referring to a project name, such as "near-cli".)

Comment (artob) [reply docs/evm/truffle.md:71]: "(Specifically `~/.near-credentials`)" => "(Specifically `~/.near-credentials`.)" (note the period)

File: docs/evm/truffle.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+## Troubleshooting
+
+If the NEAR CLI commands give the message:
+
+`Found account <yourname>.<network> is not available on the network`
+
+Please check that all the files located in `~/.near-credentials/<network>` are associated with working accounts. 
+
+To check is an account exists on the network, use the `state` command on NEAR CLI.
```
Comment (artob) [inline docs/evm/truffle.md:135]: "To check is" => "To check if"

File: docs/evm/evm-local-setup.md [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+- [Troubleshooting](#troubleshooting)
+
+---
+
+### Prerequisites
+
+- [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [Docker](https://docs.docker.com/get-docker/).
+- NodeJS version 12+. We recommend using [a version manager](https://nodejs.org/en/download/package-manager/).
+- [Python 3](https://www.python.org/download/releases/3.0/) and the [pip3 package](https://pip.pypa.io/en/stable/installing/)
```
Comment (artob) [inline docs/evm/evm-local-setup.md:34]: https://www.python.org/download/releases/3.0/ is not a useful link

File: docs/evm/evm-local-setup.md [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+    cd near-contract-helper
+
+On your terminal / command prompt:
+
+    psql
+
+at the Postgres prompt:
+
+    create user helper with superuser password 'helper';
+    create database accounts_development with owner=helper;
```
Comment (artob) [inline docs/evm/evm-local-setup.md:112]: Out of scope for this PR, but we really should not be requiring the creation of unqualified names. "helper" => "near_helper"

**Review by artob — APPROVED** [review]
Overall this looks great :+1: There are a couple of minor typos to fix, and beyond those some further nits to ponder.

In any case, this is good enough to be merged. It can be improved and iterated upon later.

Due to time constraints, I didn't try executing and following all the instructions as yet, will have to come back to that later.

File: docs/evm/near-web3-provider.md (hunk 3 above)
Comment (mikedotexe) [reply docs/evm/near-web3-provider.md:104]: Hm, I think this one I'd like to pass on. Elsewhere we refer to NEAR CLI as a tool with a proper noun kind of feel.

File: docs/evm/evm-local-setup.md (hunk 9 above)
Comment (mikedotexe) [reply docs/evm/evm-local-setup.md]: Yes, I searched for a while for a good canonical link. Searched again and didn't find exactly what I was hoping for. Went with https://www.python.org/download

File: docs/evm/truffle.md (hunk 6 above)
Comment (mikedotexe) [reply docs/evm/truffle.md]: good eye thanks

**Review by artob — APPROVED** [review]
LGTM :+1:

File: docs/evm/near-web3-provider.md (hunk 3 above)
Comment (artob) [reply docs/evm/near-web3-provider.md:104]: All right.

**Pull request merged**